cmd/all-in-one/main.go

```diff
@@ -76,7 +76,8 @@ func main() {
 	if os.Getenv(storage.SpanStorageTypeEnvVar) == "" {
 		os.Setenv(storage.SpanStorageTypeEnvVar, "memory") // other storage types default to SpanStorage
 	}
-	storageFactory, err := storage.NewFactory(storage.FactoryConfigFromEnvAndCLI(os.Args, os.Stderr))
+	storageFactory, err := storage.NewFactory(storage.FactoryConfigFromEnvAndCLI(os.Args, os.Stderr),
+		[]string{storage.KafkaStorageType})
```

Comment:
if we want to introduce optional params, we should switch to passing an Options struct instead of positional args.

https://github.com/jaegertracing/jaeger-lib/blob/3fa9aa04b65a266a1bbe0a5c8ac1408281b2a64a/metrics/factory.go#L31

Reply:
I choose to make it an optional parameter as the [FactoryConfig] is currently only for config from the Cobra, but I could indeed add an Options struct inside the same package and use that!

[FactoryConfig]: https://github.com/jaegertracing/jaeger/blob/master/plugin/storage/factory_config.go#L34-L39